Automattic/wp-calypso
# Odyssey Stats: replace config api with custom implementation

Related to #75257 

## Proposed Changes

The PR proposes to replace the `@automattic/calypso-config` with `src/lib/config-api` which,

 * Supports custom config data source, which is useful for Odyssey Apps.
 * Restores URL/Hash based feature flag gating for Odyssey Apps.
 * Must be initialized before using.
 * Keeps only one copy of configData.

Also it,
 * Adds unit test ability
 * Copies tests for `@automattic/create-calypso-config` over to make sure our implementation works the same. 
 * Adds automated test on TeamCity

As config is such a fundamental package, and is used everywhere, that I'm not keen to replace it with our implementation even though they are compatible.

Credits go to editing-toolkit for the jest config.

## Testing Instructions

* Ensure Odyssey Stats and Odyssey Widget still work okay for Jetpack bleeding edge
* Ensure Odyssey Stats and Odyssey Widget still work okay for Jetpack from https://github.com/Automattic/jetpack/pull/30147
* Run `cd apps/odyssey-stats && yarn test:js`
* Ensure all tests pass

## Pre-merge Checklist
